src/platform/plugins/shared/unified_doc_viewer/public/components/content_framework/table/components/data_grid/index.tsx

```diff
@@ -0,0 +1,159 @@
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { EuiDataGridProps } from '@elastic/eui';
+import { EuiDataGrid } from '@elastic/eui';
+import type { DocViewRenderProps } from '@kbn/unified-doc-viewer/types';
+import React, { useMemo } from 'react';
+import { css } from '@emotion/react';
+import { FieldRow } from '../../../../doc_viewer_table/field_row';
+import {
+  getFieldCellActions,
+  getFieldValueCellActions,
+} from '../../../../doc_viewer_table/table_cell_actions';
+import { getUnifiedDocViewerServices } from '../../../../../plugin';
+
+export interface DataGridField {
+  name: string;
+  value: unknown;
+  nameCellContent?: React.ReactNode;
+  valueCellContent?: React.ReactNode;
+}
+
+interface DataGridProps
+  extends Pick<
+    DocViewRenderProps,
+    'hit' | 'dataView' | 'columnsMeta' | 'filter' | 'onAddColumn' | 'onRemoveColumn' | 'columns'
+  > {
+  fields: Record<string, DataGridField>;
+  isEsqlMode: boolean;
+  title: string;
+}
+
+export const DataGrid = ({
+  hit,
+  dataView,
+  columnsMeta,
+  fields,
+  columns,
+  filter,
+  onAddColumn,
+  onRemoveColumn,
+  isEsqlMode,
+  title,
+}: DataGridProps) => {
+  const { fieldFormats, toasts } = getUnifiedDocViewerServices();
+
+  const onToggleColumn = useMemo(() => {
+    if (!onRemoveColumn || !onAddColumn || !columns) {
+      return undefined;
+    }
+    return (field: string) => {
+      if (columns.includes(field)) {
+        onRemoveColumn(field);
+      } else {
+        onAddColumn(field);
+      }
+    };
+  }, [onRemoveColumn, onAddColumn, columns]);
+
+  const rows: FieldRow[] = useMemo(
+    () =>
+      Object.entries(fields).map(([key, field]) => {
+        return new FieldRow({
+          name: key,
+          displayNameOverride: key,
+          flattenedValue: field.value,
+          hit,
+          dataView,
+          fieldFormats,
+          isPinned: false,
+          columnsMeta,
+        });
+      }),
+    [fields, hit, dataView, fieldFormats, columnsMeta]
+  );
+
+  const fieldCellActions = useMemo(
+    () => getFieldCellActions({ rows, isEsqlMode, onFilter: filter, onToggleColumn }),
+    [rows, filter, onToggleColumn, isEsqlMode]
+  );
+  const fieldValueCellActions = useMemo(
+    () => getFieldValueCellActions({ rows, isEsqlMode, toasts, onFilter: filter }),
+    [rows, toasts, filter, isEsqlMode]
+  );
+
+  const dataGridRenderCellValue = ({
+    rowIndex,
+    columnId,
+  }: {
+    rowIndex: number;
+    columnId: string;
+  }) => {
+    const fieldName = rows[rowIndex]?.name;
+    const fieldConfig = fields[fieldName];
+
+    if (!fieldConfig) return null;
+    if (columnId === 'name') return fieldConfig.nameCellContent;
+    if (columnId === 'value') return fieldConfig.valueCellContent;
+    return null;
```

Comment:
There's only two cases right?

```suggestion
    if (columnId === 'name') {
      return fieldConfig.nameCellContent;
    }
    return fieldConfig.valueCellContent;
```

Reply:
https://github.com/elastic/kibana/pull/231782/commits/b0c96f4f80344257518dac4e6b022e6332ceb84b